frontend/src/metabase/dashboard/reducers.js

```diff
@@ -372,15 +371,37 @@ const loadingDashCards = handleActions(
       }),
     },
     [FETCH_DASHBOARD_CARD_DATA]: {
-      next: (state, { payload: { currentTime, dashcardIds } }) => {
-        const loadingIds = Array.isArray(dashcardIds) ? dashcardIds : [];
+      next: (state, { payload: { currentTime, loadingIds, dashboardId } }) => {
+        // Loading a new dahsboard e.g. when navigating to a new dashboard
+        if (dashboardId !== state.dashboardId) {
```

Comment:
Previously, this block isn't needed, because we'll just always replace `loadingIds` with a new set, but with new requirements, there could be a case where we dispatch not all dashcard IDs in a single dashboard. e.g. if a dashboard has dashcard 1 and 2, and 1 is still loading when the refresh interval came, this action will only dispatch 2 in `loadingIds`, but if the new `loadingIds` are for different dashboard, we will keep the previous behavior.